src/main/java/org/breedinginsight/daos/ProgramDAO.java

```diff
@@ -124,20 +147,50 @@ public ProgramDAO(Configuration config, DSLContext dsl, BrAPIProvider brAPIProvi
         return resultPrograms;
     }
 
-    public ProgramBrAPIEndpoints getProgramBrAPIEndpoints() {
-        return new ProgramBrAPIEndpoints();
+    public int getNumProgramUsers(UUID programId) {
+        return dsl.selectCount().from(PROGRAM_USER_ROLE)
+                .where(PROGRAM_USER_ROLE.PROGRAM_ID.eq(programId)
+                        .and(PROGRAM_USER_ROLE.ACTIVE.eq(true)))
+                .fetchOne(0, Integer.class);
     }
 
     public ProgramBrAPIEndpoints getProgramBrAPIEndpoints(UUID programId) {
-        // Just returns defaults for now
-        // TODO: in future return urls for given program
+        ProgramEntity programEntity = fetchOneById(programId);
+
+        String coreUrl = defaultBrAPICoreUrl;
+        String genoUrl = defaultBrAPIGenoUrl;
+        String phenoUrl = defaultBrAPIPhenoUrl;
+
+        // only storing one program brapi url for now so set all to that one
+        if (!StringUtils.isBlank(programEntity.getBrapiUrl())) {
+            String brapiUrl = programEntity.getBrapiUrl();
+            coreUrl = brapiUrl;
+            genoUrl = brapiUrl;
+            phenoUrl = brapiUrl;
+        }
+
         return ProgramBrAPIEndpoints.builder()
-                .coreUrl(Optional.of(defaultBrAPICoreUrl))
-                .genoUrl(Optional.of(defaultBrAPIGenoUrl))
-                .phenoUrl(Optional.of(defaultBrAPIPhenoUrl))
+                .coreUrl(Optional.of(coreUrl))
+                .genoUrl(Optional.of(genoUrl))
+                .phenoUrl(Optional.of(phenoUrl))
                 .build();
     }
 
+    public boolean brapiUrlSupported(String brapiUrl) {
+        boolean supported = true;
+        brAPIClientProvider.setBrapiClient(brapiUrl);
+        ServerInfoApi serverInfoAPI = brAPIProvider.getServerInfoAPI(BrAPIClientType.BRAPI);
+
+        // for now just check for 200 response, in future we could check actual required endpoints
+        try {
+            ApiResponse<BrAPIServerInfoResponse> response = serverInfoAPI.serverinfoGet(BrAPIWSMIMEDataTypes.APPLICATION_JSON);
+        } catch (ApiException e) {
+            log.error(e.getMessage());
+            supported = false;
+        }
+        return supported;
+    }
```

Comment:
This seems like more of a service method than a dao method. I'm cool with it staying this way if you prefer it in the dao though.

Reply:
I think I put it there initially because we've been doing all the brapi API calls in the DAOs. I could see how there is some extra logic there around it that could be better in the service. If it's not a big deal I'll probably just leave it.